lib/sla.js

```diff
@@ -1,6 +1,77 @@
+/*jslint node: true */
 'use strict';
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Receives initial input and begins the application.
+ * @param {Object} client - Client received from app.js.
+ * @callback end - Handles errors and ends the application.
+ */
+module.exports = function(client, end) {
+  designateMixingBridge();
+
+  /**
+   * Looks for an existing mixing bridge and creates one if none exist.
+   */
+  function designateMixingBridge() {
+    client.bridges.list(function(err, bridges) {
+      var bridge = bridges.filter(function(candidate) {
+        return candidate.bridge_type === 'mixing';
+      })[0];
+      if (bridge) {
+        console.log('Using existing mixing bridge %s', bridge.id);
+      } else {
+        bridge = client.bridges.create({type: 'mixing'}, function(err, bridge) {
+          if (err) {
+            end(err);
+          } else {
+            console.log('Created new mixing bridge %s', bridge.id);
+          }
+        });
+      }
+      createChannel(client, bridge);
+    });
+  }
+
+  /**
+   * Adds created channel to the bridge.
+   * @param {Object} channel - Created channel to be added to the bridge.
+   * @param {Object} bridge - Bridge that channel is to be added to.
+   */
+  function addChannelsToBridge(channel, bridge) {
+    console.log('Adding channel to bridge %s', bridge.id);
+    bridge.addChannel({channel: channel.id}, function(err) {
+      if (err) {
+        channel.hangup();
+        end(err);
+      }
+    });
+    bridge.on('ChannelEnteredBridge', function(event) {
+      console.log("Channel %s has entered the bridge", event.channel.id);
+    });
+    bridge.on('ChannelLeftBridge', function(event) {
+      console.log("Channel %s has left the bridge", event.channel.id);
+      end(null);
+    });
+  }
+
+  /**
+   * Creates a channel to a specified endpoint and places it in the Stasis app.
+   * @param {Object} client - Object that contains information from the ARI
+   *   connection.
+   * @param {Object} bridge - Object that defines the bridge to be passed to
+   *   addChannelsToBridge().
+   */
+  function createChannel(client, bridge) {
+    var channel = client.Channel();
+    channel.originate({endpoint: 'SIP/phone', app: 'sla'},
+      function(err) {
+        if (err) {
+          end(err);
+        }
+      }
+    );
+    channel.on('StasisStart', function(event, outgoing) {
```

Comment:
outgoing seems vague in this case. I would consider changing the name to something a bit more descriptive.